src/python/WMCore/MicroService/Unified/MSRuleCleaner.py

```diff
@@ -50,27 +116,341 @@ def execute(self, reqStatus):
+        else:
+            # We shouldn't be here:
+            msg = "Skipping workflow: %s - "
+            msg += "Does not fall under any of the defined categories."
+            self.logger.info(msg, wflow['RequestName'])
+
+        # Archive:
+        try:
+            self.plineArchive.run(wflow)
+            self.wfCounters['archived'] += 1
+        except MSRuleCleanerArchival as ex:
+            msg = "%s: Archival Error: %s. "
+            msg += "\nWill retry again in the next cycle."
+            self.logger.error(msg, wflow['PlineMarkers'][-1], ex.message())
+        except Exception as ex:
+            msg = "%s General error from pipeline. Error: \n%s. "
+            msg += "\nWill retry again in the next cycle."
+            self.logger.exception(msg, wflow['PlineMarkers'][-1], str(ex))
+        return wflow
+
+    def setPlineMarker(self, wflow, pName=None):
+        """
+        A function intended to mark which is the pipeline currently working
+        on the workflow. It is supposed to be called always as a first function
+        in the pipeline.
+        :param  wflow:   A MSRuleCleaner workflow representation
+        :param  pName:   The name of the functional pipeline
+        :return wflow:
+        """
+        # NOTE: The current functional pipeline MUST always be appended at the
+        #       end of the 'PlineMarkers' list
+
+        # First get rid of the default:
+        if not wflow['PlineMarkers']:
+            wflow['PlineMarkers'] = []
+
+        # Then push our current value into the markers list:
+        wflow['PlineMarkers'].append(pName)
+
+        # Populate the list of flags to be used later:
+        if pName not in wflow['RulesToClean']:
+            if pName in [pline.name for pline in self.cleanuplines]:
+                wflow['RulesToClean'][pName] = []
+        if pName not in wflow['CleanupStatus']:
+            if pName in [pline.name for pline in self.cleanuplines]:
+                wflow['CleanupStatus'][pName] = False
+        return wflow
+
+    def _checkClean(self, wflow):
+        """
+        An auxiliary function used to only check the temporary cleanup status.
+        It basically takes the pipelines registered in 'PlineMarkers' that have
+        already worked on the workflow as a mask and applies this mask over
+        the set of flags in the 'CleanupStatus' field and then reduces the
+        result to a single bool value
+        """
+        # NOTE: This is one of the few functions taking a workflow as an argument
+        #       but returning a bool, since it is an auxiliary function and is not
+        #       supposed to be called as a standalone function in a pipeline.
+        # NOTE: `all([]) == True`, ergo all the 'rejected' && 'aborted-completed' workflows
+        #       are also counted as properly cleaned and can trigger archival later
+
+        # Build a list of bool flags based on the mask of PlineMarkers
```

Comment:
Please break this down into multiple lines to make it more readable.

Reply:
Again, I couldn't imagine a more readable way to express how this list of flags to be checked is build.

Reply:
It's really a matter of taste here... I tend to prefer constructions like:
```
cleanFlagsList = []
for key in wflow['PlineMarkers']:
    if key in wflow['CleanupStatus'].keys():
        cleanFlagsList.append(wflow['CleanupStatus'][key])
```

instead of multi-line list comprehensions. Anyhow, marking as resolved too.